packaging/installer/methods/source.md

```diff
@@ -0,0 +1,289 @@
+to build and run successfully:
+
+* libuuid
+* libuv version 1.0 or newer
+* zlib
+* GNU autoconf
+* GNU automake
+* GCC or Xcode (Clang is known to have issues in certain configurations, see [Using Clang](#using-clang))
+* A version of `make` compatible with GNU automake
+* Git (we use git in the build system to generate version info, don't need a full install, just a working `git show` command)
+
+Additionally, the following build time features require additional dependencies:
+
+* TLS support for the web GUI:
+    * OpenSSL 1.0.2 or newer _or_ LibreSSL 3.0.0 or newer.
+* dbengine metric storage:
+    * liblz4 r129 or newer
+    * OpenSSL 1.0 or newer (LibreSSL _amy_ work, but is largely untested).
+    * [libJudy](http://judy.sourceforge.net/)
+* Netdata Cloud support:
+    * A working internet connection
+    * A recent version of CMake
+    * OpenSSL 1.0.2 or newer _or_ LibreSSL 3.0.0 or newer.
+    * JSON-C (may be provided by the user as shown below, or by the system)
+
+## Preparing the Source Tree
+
+Certain features in Netdata require custom versions of specific libraries,
+which the the build system will link statically into Netdata. These
+libraries and their header files must be copied into specific locations
+in the source tree to be used.
+
+### Netdata Cloud
+
+Netdata Cloud functionality requires custom builds of libmosquitto and
+libwebsockets.
+
+#### libmosquitto
+
+Netdata maintains a csutom fork of libmosquitto at
+https://github.com/netdata/mosquitto which is needed for correct operation
+of Netdata Cloud functionality. To prepare this library for the build
+system:
+
+1. Verify the tag that Netdata expects to be used by checking the contents
+   of `packaging/mosquitto.version` in your Netdata sources.
+2. Obtain the sources for that version by either:
+    * Navigating to https://github.com/netdata/mosquitto/releases and
+      donwloading and unpacking the source code archive for that release.
+    * Cloning the repository with `git` and checking out the required tag.
+3. If building on a platfom other than Linux, prepare the mosquitto
+   sources by running `cmake -D WITH_STATIC_LIBRARIES:boolean=YES .` in
+   the mosquitto source directory.
+4. Build mosquitto by running `make -C lib` in the mosquitto source directory.
+5. In the Netdata source directory, create a directory called `externaldeps/mosquitto`.
+6. Copy `lib/mosquitto.h` from the mosquitto source directory to
+   `externaldeps/mosquitto/mosquitto.h` in the Netdata source tree.
+7. Copy `lib/libmosquitto.a` from the mosquitto source directory to
+   `externaldeps/mosquitto/libmosquitto.a` in the Netdata source tree. If
+   building on a platform other than Linux, the file that needs to be
+   copied will instead be named `lib/libmosquitto_static.a`, but it
+   still needs to be copied to `externaldeps/mosquitto/libmosquitto.a`.
+
+#### libwebsockets
+
+Netdata uses the standard upstream version of libwebsockets located at
```

Comment:
1 or 2 sentences explaining why we need the build option

Reply:
While not explaining thisin detail looks kind of bad, I'd argue it will actually look worse to properly explain it, as the explanation amounts to 'Because we think you need to have this feature enabled that you will probably never use and don't care enough about the larger number of _users_ (as opposed to total installs) who will not need this to make it optional.'.

The relevant option is SOCKS5 support in LWS, which we require for SOCKS5 support for the ACLK. I can almost guarantee that more than 95% of our users will never need this, but a large percentage of total installs will, as it's largely just a corporate network thing (and I'm willfully ignoring the tiny percentage of crazy people who may want to run the Netdata Cloud connection over Tor or or other anonymity networks).

Reply:
We actually had some questions about running it over tor before. Maybe just put "because we require SOCKS5 support", that should be enough.